bootstrap.py

```diff
@@ -268,8 +278,19 @@ def post_json(url, jdata):
         request.add_header("Accept", "application/json")
         request.get_method = lambda: 'POST'
         reply = opener.open(request)
+    except urllib2.HTTPError, e:
+        if e.code == 401:
```

Comment:
We may have the following (just a 2 minute brainstorming):
1) wrong request made (err 400, POST-GET(?)-DELETE)
2) wrong username/password (err 403, POST-GET-DELETE hardcoding "admin" is just wrong)
3) http refused connection (connection reset, code?)
4) firewall (connection timeout, code?)

and so on... am i wrong?